sklearn/utils/_array_api.py

```diff
@@ -48,6 +100,9 @@ def take(self, X, indices, *, axis):
             selected = [X[:, i] for i in indices]
         return self._namespace.stack(selected, axis=axis)
 
+    def isdtype(self, dtype, kind):
+        return isdtype(dtype, kind, xp=self._namespace)
+
 
 class _NumPyApiWrapper:
```

Comment:
While we are at it, maybe rename this to `_NumPyAPIWrapper` to be consistent with the casing of `_ArrayAPIWrapper`.

Or in a follow-up PR to keep the diff easier to review for the second review.

Reply:
The renaming makes sense. I opened https://github.com/scikit-learn/scikit-learn/pull/26039 to do the name change.